plugins/logging/file

```diff
@@ -275,12 +275,13 @@ sub hook_logging {
     if (   !$self->{_f}
         || !$self->{_nosplit}
         || !$transaction
+        || !UNIVERSAL::can($transaction, 'notes')
         || !$transaction->notes('file-logged-this-session'))
     {
         unless (defined $self->maybe_reopen($transaction)) {
             return DECLINED;
         }
-        $transaction->notes('file-logged-this-session', 1) if $transaction;
+        $transaction->notes('file-logged-this-session', 1) if $transaction && UNIVERSAL::can($transaction, 'notes');
```

Comment:
Please rebase this branch (git rebase smtpd/master) so that it'll merge cleanly. Also, convert this to a standard if ( ... ) { } instead of the very long line with the postfix if.